var/spack/repos/builtin/packages/llvm-amdgpu/package.py

```diff
@@ -22,9 +22,7 @@ class LlvmAmdgpu(CMakePackage):
 
     depends_on('cmake@3:', type='build')
     depends_on('python', type='build')
-    depends_on("z3")
-    depends_on("zlib")
-    depends_on("ncurses+termlib")
+    depends_on('zlib', type='link')
```

Comment:
Can you please motivate why you made this change?

Note from https://github.com/RadeonOpenCompute/llvm-project/blob/amd-stg-open/llvm/CMakeLists.txt#L349:

```
option(LLVM_ENABLE_TERMINFO "Use terminfo database if available." ON)
```
and
```
set(LLVM_ENABLE_ZLIB "ON" CACHE STRING "Use zlib for compression/decompression if available. Can be ON, OFF, or FORCE_ON")
```

and you don't disable this.

Reply:
@haampie , basically i am seeing a build failure due to z3 while building rocm-device-libs.  llvm-amdgpu will build fine with the above inclusions but rocm-device-libs is failing with the below -
>> 62    /opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld:
           cannot find -ltinfo
  >> 63    collect2: error: ld returned 1 exit status
  >> 64    make[2]: *** [utils/prepare-builtins/prepare-builtins] Error 1
     65    make[2]: Leaving directory `/tmp/root/spack-stage/spack-stage-rocm-d
           evice-libs-3.5.0-dxvh2xe47levmc3hmdedj62niaephda5/spack-build'
  >> 66    make[1]: *** [utils/prepare-builtins/CMakeFiles/prepare-builtins.dir
           /all] Error 2
     67    make[1]: Leaving directory `/tmp/root/spack-stage/spack-stage-rocm-d
           evice-libs-3.5.0-dxvh2xe47levmc3hmdedj62niaephda5/spack-build'
  >> 68    make: *** [all] Error 2

Reply:
`cannot find -ltinfo` is not `z3`, it's `ncurses`. Possibly related is https://github.com/srekolam/spack/pull/4, can you share the full linker command?

Reply:
Indeed it is... https://github.com/RadeonOpenCompute/ROCm-Device-Libs/blob/amd-stg-open/utils/prepare-builtins/CMakeLists.txt#L29 it's using the llvm support target.

Reply:
@srekolam can you merge https://github.com/srekolam/spack/pull/4 and try again?

Reply:
@haampie ,i got the patch and applied the changes manually. I see the issue is resolved. the rocm-device-libs builds with out any errors.